cmake/kokkos_backends.cmake

```diff
@@ -10,7 +10,7 @@ MACRO(CHECK_KOKKOS_BACKEND BE)
 ENDMACRO(CHECK_KOKKOS_BACKEND)
 
 CHECK_KOKKOS_BACKEND(SERIAL)
-CHECK_KOKKOS_BACKEND(PTHREAD)
+CHECK_KOKKOS_BACKEND(THREADS)
```

Comment:
This is the bit that needs to be done more carefully

Reply:
If you decide you want to be backward compatible (not suggesting you absolutely have to), you could do something like
```suggestion
# for backward compatibility.  can be dropped when requiring Kokkos 3.6
#CHECK_KOKKOS_BACKEND(THREADS)
IF (Kokkos_ENABLE_THREADS OR Kokkos_ENABLE_PTHREAD)
  SET(KOKKOS_ENABLE_THREADS ON)
  SET(KOKKOSKERNELS_INST_EXECSPACE_THREADS_DEFAULT ON)
ENDIF()
```